### ISSUE TITLE
'unicode' object has no attribute 'seek' error when trying to use the importscan api

### ISSUE THREAD

**Issue by petrabrunner** [DefectDojo/django-DefectDojo#845]
Please use the following template when submitting issues: (If this is a security issue, then review the [disclosure policy](https://github.com/DefectDojo/django-DefectDojo/blob/master/SECURITY.md) and submit via our security reporting program.

### Issue Type 
- [x ] Bug
- [ ] Enhancement

### Description
I am having problems uploading Burp scan reports automatically. 
Since I cannot investigate the api via the explorer-ui (see https://github.com/DefectDojo/django-DefectDojo/issues/844) I tried making the best of the information given in the "raw"-json response (the one you get when clicking on "Raw" when viewing the <defectdojo>/api/v1/doc/ page.

Anyway I tried using the /api/v1/importscan/ rest-endpoint, but unfortunately I was not able to get anything else than a 500 error repsonse ->> "error_message": "'unicode' object has no attribute 'seek'"

Any suggestions, what I could have done wrong? my personal guess would be, that the system cannot find the file I entered. yet. I tried in the browser (and I am starting this rest-call at the same system the file is located) and it is located at exactly the location I specified.

I tried work-around-ing it along the lines of "https://github.com/DefectDojo/django-DefectDojo/issues/521" (aka I tried to use defectdojos webui-rest endpoints (including login, and csrfmiddlewaretoken fetching)) but that did not work, because for some reason the request always ends in 403 forbidden - it seems that the csrf middleware token is not valid (although I dont understand why, since it seems that I am executing exactly the same calls the browser triggers for me when clicking in the ui) so that ended in a roadblock as well. 

At the moment I am quite stumped because nothing seems to be working... :( so any help would be appreciated.

### Operating System 
Please include the version number (ex: Ubuntu 16.04)
Ubuntu 18.04 LTS

### Install

- [ ] setup.bash
- [ x] docker

### DefectDojo Version
latest master (latest commit: https://github.com/DefectDojo/django-DefectDojo/commit/f4dd2aaedcbda22b501f07a06e11240c6c5e85de)

### Steps to Reproduce
api/v1/ -calls:

---

POST /api/v1/importscan/ HTTP/1.1
Host: <host>
Content-Type: application/json
Accept-Encoding: gzip, deflate
Authorization: ApiKey <user>:<key>
Connection: close
Content-Length: 317

{"minimum_severity":"Info","scan_date":"2019-01-10","verified":"True","file":"file:///C:/pentest/scanReports/ScanReport.xml","lead":"/api/v1/users/1/","tags":"burpscan","active":"True","engagement":"/api/v1/engagements/22/","scan_type":"Burp Scan"}

---

unfortunately the response was a hurdle for me. So far I could not workaround the 500 error.

---

HTTP/1.1 500 Internal Server Error
Server: nginx
Date: Fri, 11 Jan 2019 08:07:56 GMT
Content-Type: application/json
Content-Length: 1307
Connection: close
Vary: Cookie
X-Frame-Options: SAMEORIGIN

{"error_message": "'unicode' object has no attribute 'seek'", "traceback": "Traceback (most recent call last):\n\n  File \"/usr/local/lib/python2.7/dist-packages/tastypie/resources.py\", line 228, in wrapper\n    response = callback(request, *args, **kwargs)\n\n  File \"/usr/local/lib/python2.7/dist-packages/tastypie/resources.py\", line 468, in dispatch_list\n    return self.dispatch('list', request, **kwargs)\n\n  File \"/usr/local/lib/python2.7/dist-packages/tastypie/resources.py\", line 500, in dispatch\n    response = method(request, **kwargs)\n\n  File \"/usr/local/lib/python2.7/dist-packages/tastypie/resources.py\", line 1409, in post_list\n    updated_bundle = self.obj_create(bundle, **self.remove_api_resource_names(kwargs))\n\n  File \"/opt/django-DefectDojo/dojo/api.py\", line 1356, in obj_create\n    parser = import_parser_factory(bundle.data['file'], t)\n\n  File \"/opt/django-DefectDojo/dojo/tools/factory.py\", line 37, in import_parser_factory\n    parser = BurpXmlParser(file, test)\n\n  File \"/opt/django-DefectDojo/dojo/tools/burp/parser.py\", line 44, in __init__\n    tree = self.parse_xml(xml_output)\n\n  File \"/opt/django-DefectDojo/dojo/tools/burp/parser.py\", line 65, in parse_xml\n    xml_file.seek(0)\n\nAttributeError: 'unicode' object has no attribute 'seek'\n"}

-------------------------------------------------------

**Comment by alekbc** [comment]
Hi, this may be related to #819 . it may worth to take a look

**Comment by petrabrunner** [comment]
hi, @alekbc  thx for the hint, you pointed me in the right direction.
Seems my problem was, that the file referenced by the "file" attribute has to be either copied onto the dojo-host, or be reachable from the dojo-host... (so served by some webserver, or networkshare, or whatever...) that is what I was missing. (kinda logical but whatever)

to workaround the error I was getting I adapted the parse_xml method slightly:

```
 def parse_xml(self, xml_file):
        [...]     
        tree = None
        try:
            tree = etree.parse(xml_file)
        except Exception, e:
            print "here:"
            print str(e)
            print "ended"
            # Solution to remove unicode characters in xml, tried several
            #xml_file.seek(0)
            file123 = open(xml_file, "r")
            data = file123.readlines()
            file123.close()
            printable = set(string.printable)
            data = filter(lambda x: x in printable, data)
            tree = etree.fromstring(data, etree.XMLParser(encoding='ISO-8859-1', ns_clean=True, recover=True))

    return tree
```
basically I uncommented the xml_file.seek(0) line. - which caused the weird error. after that it simply told me that the file could not be found - which in turn pointed me towards the solution. yea...
